Automattic/jetpack
# Rewrite monorepo install script

While working through the setup docs, I tried to use the install script for the first time, and ran into dozens of hiccups on Mac on Linux. It also previously ran a lot of extra commands when unneeded, and didn't consistently set up the environment across shells. I went ahead and rewrote the install script; it's a bit verbose, but should be a bit more usable.

I don't account for all edge cases, but it should be robust enough, and should readily exit if something looks broken.

Given we run this in as an executable (i.e. in a subshell), restarting the terminal is required if environmental vars are changed. This was always the case.

## Does this pull request change what data or activity we track or use?

Nope.

## Testing instructions:






If you already have everything set up, you should be able to run the script and it should run through without any needed changes.

However, feel free to try on a stock macOS or Debian-based distro and ensure things run smoothly.
